routes/shopItemByStatus.js

```diff
@@ -1,25 +1,51 @@
 const express = require("express");
 const utility = require("../utility.js");
+const db = require("../server.js");
 
 findStatusRouter = express.Router();
 
 findStatusRouter.use(utility.readDatabaseMiddleware);
 
+//The original code
+// findStatusRouter.get("/", (req, res, next) => {
+//     const isCompleted = req.query.isCompleted || "";
+//     let groceryArray = req.database.shopItems;
+//     if (isCompleted === "true" || isCompleted === "false" || isCompleted === "") {
+//         if (isCompleted === "true") {
+//             groceryArray = utility.filterByStatus(groceryArray, true)
+//         } else if (isCompleted === "false") {
+//             groceryArray = utility.filterByStatus(groceryArray, false)
+//         }
+//         res.status(200);
+//         res.send(groceryArray);
+//     } else {
+//         res.status(400).send();
+//     }
+
+// });
+
+//My solution with sqlite
 findStatusRouter.get("/", (req, res, next) => {
-    const isCompleted = req.query.isCompleted || "";
-    let groceryArray = req.database.shopItems;
-    if (isCompleted === "true" || isCompleted === "false" || isCompleted === "") {
-        if (isCompleted === "true") {
-            groceryArray = utility.filterByStatus(groceryArray, true)
-        } else if (isCompleted === "false") {
-            groceryArray = utility.filterByStatus(groceryArray, false)
+    db.all("SELECT rowid, * FROM grocery_list", (err, row) => {
+        if (err) {
+            throw err
+        } else {
+            const isCompleted = req.query.isCompleted || "";
+            if (isCompleted === "true" || isCompleted === "false" || isCompleted === "") {
+                if (isCompleted === "true") {
+                    row = utility.filterByStatus(row, Number(1))
+                } else if (isCompleted === "false") {
+                    row = utility.filterByStatus(row, Number(0))
+                }
```

Comment:
Instead of filtering here, try use SQL for filtering. When `isCompleted` is "true" or "false" you can just use a WHERE clause:

Something like:

```js
if (isCompleted === "true" || isCompleted === "false") {
  db.all("SELECT id, name, is_complete FROM grocery_list WHERE is_complete=?", [isCompleted], (err, rows) => {
    // ... rest of the code, rows already filtered
  }
}
else {
  db.all("SELECT id, name, is_complete FROM grocery_list", [], (err, rows) => {
    // ... rest of the code, rows contains everything 
  }
}
```

There are a few ways to write one select only to do both but I think the above is fine as well.